rsocket-core/src/commonTest/kotlin/io/rsocket/kotlin/frame/io/CustomMimeTypeTest.kt

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2015-2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.rsocket.kotlin.frame.io
+
+import io.ktor.utils.io.core.*
+import io.rsocket.kotlin.core.*
+import kotlin.test.*
+
+class CustomMimeTypeTest {
+    val name = "message/x.foo"
+    private val fooMimeType = CustomMimeType(name)
+
+    @Test
+    fun customMimeTypeSerialization() {
```

Comment:
While we are here, It would be nice to add several tests for `custom` mime type encoding - when length of mime type is 1, 128 and something in between - to test that min/max lengths are properly supported.